src/plugins/intel_cpu/src/edge.cpp

```diff
@@ -65,22 +65,39 @@ void Edge::drop() {
     _drop_from(getChild()->parentEdges);
 }
 
-void Edge::collectConsumers(std::vector<NodePtr>& result) const {
+void Edge::collectConsumers(std::vector<NodePtr>& result, bool nested) const {
     if (!this->getChild()->getChildEdges().empty() && this->inPlace(LOOK_DOWN)) {
         if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
             auto peerOutputNum = this->getOutputNum();
             auto peerInPlacePort = peerChildSPD->getConfig().inConfs[peerOutputNum].inPlace();
             auto& vecChildEdges = this->getChild()->getChildEdgesAtPort(peerInPlacePort);
             for (auto childEdge : vecChildEdges) {
-                childEdge->collectConsumers(result);
+                childEdge->collectConsumers(result, true);
             }
         }
-    } else {
-        auto childNode = this->getChild();
-        if (Type::ShapeOf == childNode->getType()) {
-            // ShapeOf doesn't actually read the data, it only reads shape
-            return;
+        return;
+    } else if (!this->getChild()->getChildEdges().empty()) {
+        if (auto peerChildSPD = this->getChild()->getSelectedPrimitiveDescriptor()) {
+            bool isBaseEdge = false;
+            for (size_t i = 0; i < peerChildSPD->getConfig().outConfs.size(); i++) {
+                auto peerOutPlacePort = peerChildSPD->getConfig().outConfs[i].inPlace();
+                if (peerOutPlacePort == this->getOutputNum()) {
+                    isBaseEdge = true;
+                    for (auto childEdge : this->getChild()->getChildEdgesAtPort(i))
+                        childEdge->collectConsumers(result, true);
+                }
+            }
+            if (isBaseEdge)
```

Comment:
a minor suggestion - change isBaseEdge to some other names like beInplaced or beReference. BaseEdge has a specific meaning in graph subsystem and easy to result to misunderstanding here.

Reply:
done